src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -948,7 +948,8 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                         return semanticModel;
                     });
 
-                    _compilation.EventQueue.Enqueue(new SymbolDeclaredCompilationEvent(_compilation, methodSymbol, lazySemanticModel));
+                    MethodSymbol symbolToProduce = methodSymbol.PartialDefinitionPart ?? methodSymbol;
+                    _compilation.EventQueue.Enqueue(new SymbolDeclaredCompilationEvent(_compilation, symbolToProduce, lazySemanticModel));
```

Comment:
You have to give the compilation event a syntax reference in order to get back a semantic model, so it will only give you this one if you're asking about the implementation part. Since the declaration has *locations* for both parts, this is the one we need to cache to get the best performance.